src/plugins/intel_gpu/tests/test_cases/serialization_test.cpp

```diff
@@ -0,0 +1,127 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "test_utils.h"
+#include "intel_gpu/graph/serialization/binary_buffer.hpp"
+#include "intel_gpu/graph/serialization/layout_serializer.hpp"
+#include "intel_gpu/graph/serialization/string_serializer.hpp"
+
+using namespace cldnn;
+using namespace ::tests;
+
+TEST(serialization_gpu, layout_from_string) {
+    membuf mem_buf;
+    {
+        std::ostream out_mem(&mem_buf);
+        BinaryOutputBuffer ob = BinaryOutputBuffer(out_mem);
+
+        std::stringstream ss;
+        ss << InferenceEngine::Layout::ANY;
+        ob << ss.str();
+
+        ss.str("");
+        ss << InferenceEngine::Layout::NCHW;
+        ob << ss.str();
+
+        ss.str("");
+        ss << InferenceEngine::Layout::NHWC;
+        ob << ss.str();
+
+        ss.str("");
+        ss << InferenceEngine::Layout::NCDHW;
+        ob << ss.str();
+
+        ss.str("");
+        ss << InferenceEngine::Layout::NDHWC;
+        ob << ss.str();
+
+        ss.str("");
+        ss << InferenceEngine::Layout::OIHW;
+        ob << ss.str();
+
+        ss.str("");
+        ss << InferenceEngine::Layout::GOIHW;
+        ob << ss.str();
+
+        ss.str("");
+        ss << InferenceEngine::Layout::OIDHW;
+        ob << ss.str();
+
+        ss.str("");
+        ss << InferenceEngine::Layout::GOIDHW;
+        ob << ss.str();
+
+        ss.str("");
+        ss << InferenceEngine::Layout::SCALAR;
+        ob << ss.str();
+
+        ss.str("");
+        ss << InferenceEngine::Layout::C;
+        ob << ss.str();
+
+        ss.str("");
+        ss << InferenceEngine::Layout::CHW;
+        ob << ss.str();
+
+        ss.str("");
+        ss << InferenceEngine::Layout::HWC;
+        ob << ss.str();
+
+        ss.str("");
+        ss << InferenceEngine::Layout::HW;
+        ob << ss.str();
+
+        ss.str("");
+        ss << InferenceEngine::Layout::NC;
+        ob << ss.str();
+
+        ss.str("");
+        ss << InferenceEngine::Layout::CN;
+        ob << ss.str();
+
+        ss.str("");
+        ss << InferenceEngine::Layout::BLOCKED;
+        ob << ss.str();
+    }
+    {
+        std::istream in_mem(&mem_buf);
+        BinaryInputBuffer ib = BinaryInputBuffer(in_mem, get_test_engine());
+    
+        std::string layout;
+        ib >> layout;
+        ASSERT_EQ(cldnn::layout_from_string(layout), InferenceEngine::Layout::ANY);
```

Comment:
better to use EXPECT here, since if case of several problems in tests - all of them will be shown.
Assert shows only the first one

Or maybe even better to write a parametrized tests, where parameter - layout values.

Reply:
I replaced `ASSERT_EQ` with `EXPECT_EQ` as you guided. Thank you!